### ISSUE TITLE
mempoolminfee returns incorrect data before the dynamic mempool limit has changed

### ISSUE THREAD

**Issue by gmaxwell** [bitcoin/bitcoin#6941]
"mempoolminfee": 0.00000000

When the minrelaytxfee is > 0.

**Comment by TheBlueMatt** [comment]
This is expected (the mempool's min fee is 0), though maybe we could change it to just minfee and use max of the two...

**Comment by gmaxwell** [comment]
This is confusing, since the _effective_ min feerate is not zero, it's the minrelayfee. perhaps if there were just another field to show the other data.

**Comment by sipa** [comment]
The fact there there is a "mempool minrelay fee" different from the conceptual minimum relay fee is an implementation detail; no need to expose it through RPC.

**Comment by TheBlueMatt** [comment]
@sipa ACK, it should be one unified field, not two.

**Comment by MarcoFalke** [comment]
> no need to expose it through RPC

At least mention it in `help getmempoolinfo`... ... Another reason to clean up the rpc "hard coded white space mess".

**Comment by morcos** [comment]
@sipa for now they are different in whether or not priority (free) txs are allowed.

**Comment by MarcoFalke** [comment]
An attempt to document this was done in #7118.

Anything left to do here?

**Comment by laanwj** [comment]
#7118 adds the following documentation;

> \"mempoolminfee\": xxxxx       (numeric) Minimum fee for tx to be accepted\n"

I don't think that's particularly helpful in distinguishing mempool versus relaying minimum.

> The fact there there is a "mempool minrelay fee" different from the conceptual minimum relay fee is an implementation detail; no need to expose it through RPC.

Not sure about this - `getmempoolinfo` returns info about the mempool. Relaying is an another concern - we split off the different RPC info calls for good reason. Not sure whether they should be muddled together, unless we logically make `minrelaytxfee` part of the mempool logic.

**Comment by morcos** [comment]
@laanwj the relay and mempool logic are completely intertwined and both numbers affect both.
I actually don't think we should change anything right now (other than potentially better documentation)

I think its useful to be aware of both numbers.
The `minrelaytxfee` is a fixed number, but txs can get in below that subject to free rate limiting.
The `mempoolminfee` is a floating number below which txs can not get in period.
Even if we eliminate free transactions entirely, it could be useful to know the `minrelaytxfee` separately from the current `mempoolminfee`.

**Comment by laanwj** [comment]
Yes, makes sense to add `minrelaytxfee` to the same RPC then, then put that explanation in the RPC help.

**Comment by jonathancross** [comment]
Has been any progress in adding `minrelaytxfee` to `getmempoolinfo` RPC?

**Comment by MarcoFalke** [comment]
@jonathancross Not yet. Pull request welcome :)

**Comment by jeffrade** [comment]
Any reason to keep this issue open only for the doc changes when calling `help getmempoolinfo`?  I'd be happy to make the doc changes in another PR (once #11475 is merged), but can they be simply be made in [#11475](https://github.com/bitcoin/bitcoin/pull/11475)?
